src/Sylius/Component/Core/Uploader/ImageUploader.php

```diff
@@ -14,67 +14,57 @@
 use Gaufrette\Filesystem;
 use Sylius\Component\Core\Model\ImageInterface;
 
-class ImageUploader implements ImageUploaderInterface
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ */
+final class ImageUploader implements ImageUploaderInterface
 {
     /**
      * @var Filesystem
      */
-    protected $filesystem;
+    private $filesystem;
+
+    /**
+     * @var PathGeneratorInterface
+     */
+    private $pathGenerator;
 
     /**
      * @param Filesystem $filesystem
+     * @param PathGeneratorInterface $pathGenerator
      */
-    public function __construct(Filesystem $filesystem)
+    public function __construct(Filesystem $filesystem, PathGeneratorInterface $pathGenerator)
     {
         $this->filesystem = $filesystem;
+        $this->pathGenerator = $pathGenerator;
     }
 
     /**
      * {@inheritdoc}
      */
     public function upload(ImageInterface $image)
```

Comment:
Sure, I would also like to rename `ImageInterface` to `FileInterface` - there's one project, where I extend Sylius' `ImageInterface` in order to upload CSS files, but with reusing the same logic.